packages/gatsby-theme-newrelic/src/components/CookieConsentDialog.js

```diff
@@ -0,0 +1,149 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import Cookies from 'js-cookie';
+import { css } from '@emotion/core';
+import ExternalLink from './ExternalLink';
+import Button from './Button';
+
+const CookieConsentDialog = ({
+  cookieName,
+  handleCookieConsent
```

Comment:
Do we need to expose a handler here for when the cookie consent is accepted or denied? I think this component should do as much of the heavy lifting as possible. That would include: reading from the cookie for the initial state, setting the cookie when the user chooses to accept/deny the terms, and initiating the tracking code if the user decides to accept the cookie consent. To me, this is one of those components that makes sense to be very opinionated because the cookie consent dialog itself is very opinionated to New Relic sites (regular Joe is not going to be installing this theme and using this component out-of-the-box for example).

I think for now, it makes sense to remove this handler and add it back when we have a use case for it. Until then, lets try and do a lot of the heavy lifting here.

Reply:
Okay! Yeah that was the thing I was deciding between and I wasn't sure how we wanted to handle it. I didn't know if we wanted to be able to configure how the cookie consent is handled. If we're sure we want google analytics on each site that uses the theme, then I'm definitely pro letting less configuration and having the component be very opinionated.

Reply:
I think it's good to make it very opinionated now, and then to change later in case we need to support different handling of the cookies.